src/coreclr/System.Private.CoreLib/src/System/Reflection/MdImport.cs

```diff
@@ -298,117 +349,166 @@ public void EnumEvents(int mdTypeDef, out MetadataEnumResult result)
             Enum(MetadataTokenType.Event, mdTypeDef, out result);
         }
 
+        #region FCalls
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern string? _GetDefaultValue(IntPtr scope, int mdToken, out long value, out int length, out int corElementType);
-        public string? GetDefaultValue(int mdToken, out long value, out int length, out CorElementType corElementType)
-        {
-            string? stringVal = _GetDefaultValue(m_metadataImport2, mdToken, out value, out length, out int _corElementType);
-            corElementType = (CorElementType)_corElementType;
-            return stringVal;
+        private static extern unsafe int GetDefaultValue(
+            IntPtr scope,
+            int mdToken,
+            out long value,
+            out byte* stringMetadataEncoding,
+            out int length,
+            out int corElementType);
+        public unsafe string? GetDefaultValue(int mdToken, out long value, out int length, out CorElementType corElementType)
+        {
+            int hr = GetDefaultValue(m_metadataImport2, mdToken, out value, out byte* stringMetadataEncoding, out length, out int corElementTypeRaw);
+            if (hr < 0)
+            {
+                throw new BadImageFormatException();
+            }
```

Comment:
```suggestion
            ThrowBadImageExceptionForHR(GetDefaultValue(m_metadataImport2, mdToken, out value, out byte* stringMetadataEncoding, out length, out int corElementTypeRaw));
```
Add a helper method for the HResult checks?

Reply:
I really dislike these chained function calls in C#. It distracts from what is actually the logic.

Reply:
I'm going to change it, but in general I find this pattern in C# to create far more noise than clarity. Especially in the above example where the name of the "helper" has more visual prominance than what is doing the work `GetDefaultValue()`.